woocommerce/woocommerce-gutenberg-products-block
# Move cart widget tests to playwright and ensure classic/block theme configs are executed correctly

This PR is based on https://github.com/woocommerce/woocommerce-blocks/pull/10663 to include the changes to the updated folder structure and newly migrated tests.

## What

This PR removes the "in the widget editor" test from the puppeteer environment and reimplements it in playwright. It also:
- Adds a `closeModalByName` util which will take a modal name and close it if it's visible.
- Ensures the correct theme is set up when running the classic/block theme tests
- Ensures the configuration files for each environment run correctly.

**It also adds a new configuration to the e2e test environment**

This is required because there are tests that rely on a specific type of theme (classic theme vs. block theme) and these cannot run in parallel because certain features are missing depending on what type of theme you're using.

Therefore, I created a new configuration for playwright which runs these classic-theme tests, added a new configuration to the GitHub action for playwright, and added new scripts in `package.json` which allow these classic theme tests to be run in isolation.

## Why

This enables us to remove one of the puppeteer test files as it would be fully migrated!

It also changes the configuration files because this is necessary to ensure the environment setup runs correctly. Without this change, running tests against the classic theme may not use the correct theme.

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Ensure all e2e tests pass, playwright and puppeteer.
2. Locally, run `npm run test:e2e -- tests/e2e/tests/cart/cart-block.merchant.classic_theme.spec.ts` then go to `localhost:8889` and verify it's using storefront.
3. Then run `npm run test:e2e -- tests/e2e/tests/cart/cart-block.merchant.block_theme.spec.ts` and go to `localhost:8889` and verify it's using twentytwentythree


## Screenshots or screencast

n/a

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Skipping
